lib/beacon/config.ex

```diff
@@ -298,16 +314,59 @@ defmodule Beacon.Config do
       publish_page: get_in(opts, [:lifecycle, :publish_page]) || []
     ]
 
+    asset_formats = Keyword.get(opts, :asset_formats, @default_asset_formats)
+
+    assigned_assets = Keyword.get(opts, :assets, [])
+
+    assets =
+      Enum.reduce(
+        asset_formats,
+        assigned_assets,
+        fn mime_type, acc ->
+          if [] == MIME.extensions(mime_type) do
+            raise(Beacon.LoaderError, "Unknown MIME type: #{mime_type}")
+          end
+
+          if mime_type_has_config(assigned_assets, mime_type) do
+            acc
+          else
+            acc ++ [{mime_type, [{:backends, [Beacon.Admin.MediaLibrary.Provider.Repo]}]}]
+          end
+        end
+      )
+      |> Enum.map(fn {mime_type, config} -> {mime_type, Keyword.put_new(config, :validations, [])} end)
+
     opts =
       opts
       |> Keyword.put(:template_formats, template_formats)
       |> Keyword.put(:lifecycle, lifecycle)
+      |> Keyword.put(:asset_formats, asset_formats)
+      |> Keyword.put(:assets, assets)
 
     struct!(__MODULE__, opts)
   end
 
   @spec fetch!(Beacon.Types.Site.t()) :: t()
-  def fetch!(site) do
+  def fetch!(site) when is_atom(site) do
     Registry.config!(site)
   end
+
+  @spec config_for_mime_type(Beacon.Config.t(), MIME.t()) :: term()
```

Comment:
you could move `Enum.into(%{})` into `config_for_mime_type/2`
```suggestion
  @spec config_for_mime_type(Beacon.Config.t(), MIME.t()) :: map()
```

Reply:
I guess the question here is, this function currently returns data of a type encoded by this module and do we want to return something else? I think when I originally wrote this, I hadn't created the `mime_config` type.

So it should be ```@spec config_for_mime_type(Beacon.Config.t(), MIME.t()) :: mime_config()```
unless you really want to return a map there.

Reply:
> unless you really want to return a map there.

Nope. It was just to avoid forcing the caller to transform the returned data.